### PR TITLE
Jm/syn 21

### PR DESCRIPTION
**Pull request by johntmyers**
- Update base TF version to 2.3+
- Add early stopping and best model saving options. Defaulted on.
- Update model history table to include DP data, if DP is enabled.
- Updated model history table to include a column that flags which epoch / model / DP info is the final model that was saved

### CODE REVIEW

File: examples/launch_synthetics.ipynb [hunk 1]
```diff
@@ -224,7 +221,7 @@
     "# num_lines=500 will override the synthetic config ``num_lines``, set whatever number you need\n",
     "# max_invalid=5000 will override the default invalid line limit that terminates execution, set whatever number you need\n",
     "\n",
-    "bundle.generate()"
+    "bundle.generate(num_lines=100)"
```
Comment (zredlined) [inline examples/launch_synthetics.ipynb:224]: I'd set num_lines=1000, and max_invalid=1000 here as default

File: examples/synthetic_records.ipynb [hunk 2]
```diff
@@ -102,7 +98,7 @@
     "    else:\n",
     "        raise Exception('record not 6 parts')\n",
     "        \n",
-    "for line in generate_text(config, line_validator=validate_record):\n",
+    "for line in generate_text(config, line_validator=validate_record, num_lines=100):\n",
```
Comment (zredlined) [inline examples/synthetic_records.ipynb:101]: I'd set num_lines=1000

File: src/gretel_synthetics/config.py [hunk 3]
```diff
@@ -114,7 +130,10 @@ class BaseConfig:
 
     # Training configurations
     max_lines: int = 0
-    epochs: int = 15
+    epochs: int = 250
```
Comment (zredlined) [inline src/gretel_synthetics/config.py:133]: Does not match the default value described in the docstring above (150). I'd recommend 100 on the top-end as a setting for epochs as you'll likely be overfitting the model at that point

File: src/gretel_synthetics/config.py [hunk 4]
```diff
@@ -135,7 +154,7 @@ class BaseConfig:
 
     # Diff privacy configs
     dp: bool = False
-    dp_learning_rate: float = 0.015
+    dp_learning_rate: float = 0.001
```
Comment (zredlined) [inline src/gretel_synthetics/config.py:157]: Let's use 0.01 as a setting, which matches the hard-coded learning rate for non-DP operation w/RMSprop. Will update as we optimize hyper params and optimizers in 15.x

**Review by zredlined — CHANGES_REQUESTED** [review]
